functions/Restore-DbaBackup.ps1

```diff
@@ -0,0 +1,215 @@
+function Restore-DbaBackup
+{
+<#
+.SYNOPSIS 
+Restores a SQL Server Database from a set of backupfiles
+
+.DESCRIPTION
+Scans a given folder for Full, Differential and Log backups. 
+OR
+Takes a set of folder arrays and processes them for Full, Differential and Log backups.
+
+These are then filtered and restored to a specified SQL Server intance and file location
+
+The backup LSN chain and RecoveryForkID will also be checked to ensure the restore is valid
+
+It can also generate restore scripts, both as part of a restore or as it's only action
+
+.PARAMETER SqlServer
+The SQL Server instance. 
+
+.PARAMETER SqlCredential
+Allows you to login to servers using SQL Logins as opposed to Windows Auth/Integrated/Trusted. 
+
+.PARAMETER Path
+Path to SQL Server backup file. This can be a full, differential or log backup file.
+
+.PARAMETER Files
+A Files object(s) containing SQL Server backup files. 
+
+.PARAMETER RestoreLocation
+Path to restore the SQL Server backups to on the target inance
```

Comment:
Ignore previous. Not had enough coffee yet and didn't read the code properly.

Proposing to base it on this logic
D:0 l:0 - Files put back where they came from
D:1 L:0 - All Files put into the specified D folder (what we have now)
D:1 L:1 - All data files into D, All log files into L
D:0 L:1 - Data Files put where they came from, Log files into L